core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -992,7 +992,7 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
 
     // This is a hack to enforce loading hdfs-site.xml.
     // See SPARK-11227 for details.
-    FileSystem.get(new URI(path), hadoopConfiguration)
+    FileSystem.get(new Path(path).toUri, hadoopConfiguration)
```

Comment:
As it is known that is hacky and ugly, maybe we can make this separate to another issue (although I am careful to say this)?